### PR TITLE
VIH-7410 fix edit interpreter issue

### PR DESCRIPTION
**Pull request by RocheL**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-7410
### Change description ###
 - fix to save change in linked participant, from summary screen.
**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts [hunk 1]
```diff
@@ -654,12 +654,20 @@ export class AddParticipantComponent extends BookingBaseComponent implements OnI
...
         if (newParticipant.hearing_role_name.toLowerCase() === HearingRoles.INTERPRETER) {
             if (this.editMode) {
-                const linkedParticipant = newParticipant.linked_participants[0];
-                const interpretee = this.hearing.participants.find(p => p.id === linkedParticipant.linkedParticipantId);
-                interpretee.linked_participants = [];
-                linkedParticipant.linkedParticipantId = this.getInterpreteeId(newParticipant.interpreterFor);
-                linkedParticipant.participantId = newParticipant.id;
-                _linkedParticipants.push(linkedParticipant);
+                if (this.localEditMode) {
+                    const linkedParticipant = newParticipant.linked_participants[0];
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts:658]: 3 nested if-statements is a bit of a code smell. recent PRs have been failing due to the cognitive complexity growing between each PRs. Could you introduce some private methods here to break this down, please? Thank you

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts [hunk 2]
```diff
@@ -651,25 +651,44 @@ export class AddParticipantComponent extends BookingBaseComponent implements OnI
...
+        let _linkedParticipants: LinkedParticipantModel[] = [];
+        if (this.editMode) {
+            _linkedParticipants = this.updateNewParticipantToLinkedParticipant(newParticipant);
+        } else {
+            _linkedParticipants = this.addNewParticipantToLinkedParticipant(newParticipant);
+        }
+        return _linkedParticipants;
+    }
+    private updateNewParticipantToLinkedParticipant(newParticipant: ParticipantModel): LinkedParticipantModel[] {
+        let _linkedParticipants: LinkedParticipantModel[] = [];
```
Comment (phillipjchambers) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts:670]: You could just do a return here without the let

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts [hunk 3]
```diff
@@ -651,25 +651,44 @@ export class AddParticipantComponent extends BookingBaseComponent implements OnI
...
+        if (this.localEditMode) {
+            const linkedParticipant = newParticipant.linked_participants[0];
+            const interpretee = this.hearing.participants.find(p => p.id === linkedParticipant.linkedParticipantId);
+            interpretee.linked_participants = [];
+            linkedParticipant.linkedParticipantId = this.getInterpreteeId(newParticipant.interpreterFor);
+            linkedParticipant.participantId = newParticipant.id;
+            _linkedParticipants.push(linkedParticipant);
```
Comment (phillipjchambers) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts:677]: return [linkedParticipant];

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts [hunk 4]
```diff
@@ -651,25 +651,44 @@ export class AddParticipantComponent extends BookingBaseComponent implements OnI
...
+        } else {
+            _linkedParticipants = this.addNewParticipantToLinkedParticipant(newParticipant);
```
Comment (phillipjchambers) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts:679]:
```
return this.addNewParticipantToLinkedParticipant(newParticipant);
```

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts [hunk 5]
```diff
@@ -651,25 +651,44 @@ export class AddParticipantComponent extends BookingBaseComponent implements OnI
...
         }
         return _linkedParticipants;
     }
+    private addNewParticipantToLinkedParticipant(newParticipant: ParticipantModel): LinkedParticipantModel[] {
+        const _linkedParticipants: LinkedParticipantModel[] = [];
```
Comment (phillipjchambers) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts:684]: as above

**Review by phillipjchambers — APPROVED** [review]

**Pull request merged**